### PR TITLE
Add pagination to the internal end-point listing deployments

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

File: api/http/api_deployments.go [hunk 1]
```diff
@@ -1093,15 +1093,24 @@ func (d *DeploymentsApiHandlers) DeploymentsPerTenantHandler(w rest.ResponseWrit
 	tenantID := r.PathParam("tenant")
```
Comment (alfrunes) [inline api/http/api_deployments.go:1093]: This entire handler could simply be:
```
func (d *DeploymentsApiHandlers) DeploymentsPerTenantHandler(w rest.ResponseWriter, r *rest.Request) {
	r.Request = r.WithContext(identity.WithContext(
		r.Context(),
		&identity.Identity{Tenant: r.PathParam("tenant")},
	))
	d.LookupDeployment(w, r)
}
```

**Review by alfrunes — APPROVED** [review]
LGTM
Though we could just reuse the exposed endpoint handler.

**Review by kjaskiewiczz — APPROVED** [review]

File: api/http/api_deployments.go (hunk 1 above)
Comment (tranchitella) [reply api/http/api_deployments.go:1090]: https://github.com/mendersoftware/deployments/pull/514/commits/8299bed94ead03d7d22186ec0b3907119b5d9516

**Comment by tranchitella** [comment]
I'm cancelling the QA pipeline, it was already green before, and the refactoring is 100% covered by unit tests.

**Pull request merged**